perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_login_namespace_researcher_access_persistent_shadow.java

```diff
@@ -1,48 +1,113 @@
 package cz.metacentrum.perun.core.impl.modules.attributes;
 
-import cz.metacentrum.perun.core.implApi.modules.attributes.UserPersistentShadowAttributeWithConfig;
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.ExtSource;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.UserExtSource;
+import cz.metacentrum.perun.core.api.exceptions.ExtSourceNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.UserExtSourceExistsException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.impl.modules.ModulesConfigLoader;
+import cz.metacentrum.perun.core.impl.modules.ModulesYamlConfigLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.UUID;
 
 /**
  * Class for checking logins uniqueness in the namespace and filling researcher-access-persistent id.
  * It is only storage! Use module login researcher-access_persistent for access the value.
  *
  */
 public class urn_perun_user_attribute_def_def_login_namespace_researcher_access_persistent_shadow
-	extends UserPersistentShadowAttributeWithConfig {
+	extends urn_perun_user_attribute_def_def_login_namespace {
 
-	private final static String attrNameResearcherAccess = "login-namespace:researcher-access-persistent-shadow";
+	private final static Logger log = LoggerFactory.getLogger(
+		urn_perun_user_attribute_def_def_login_namespace_umbrellaid_persistent_shadow.class);
 
 	private final static String CONFIG_EXT_SOURCE_NAME_RESEARCHER_ACCESS = "extSourceNameResearcherAccess";
 	private final static String CONFIG_DOMAIN_NAME_RESEARCHER_ACCESS = "domainNameResearcherAccess";
+	private final static String FRIENDLY_NAME = "login-namespace:researcher-access-persistent-shadow";
+	private final static String FRIENDLY_NAME_PARAMETER = "researcher-access-persistent-shadow";
 
-	@Override
-	public String getExtSourceConfigName() {
-		return CONFIG_EXT_SOURCE_NAME_RESEARCHER_ACCESS;
-	}
+	private final ModulesConfigLoader loader = new ModulesYamlConfigLoader();
 
+	/**
+	 * Format is: "[uuid]@{getDomainName()}" where [hash] represents a version 1 UUID (randomly generated).
```

Comment:
In the commit you mention its version 4 UUID, but javadoc states version 1, so which is it?

Reply:
Fixed